### PR TITLE
Tweak logic for msgraph catalog ingesting for display names with security groups

### PR DESCRIPTION
**Pull request by timja**
Signed-off-by: Tim Jacomb <tim.jacomb@hmcts.net>

## Hey, I just made a Pull Request!

Fixes https://github.com/backstage/backstage/issues/7888

I'd like to do a bit more testing before moving this out of draft, I'm having trouble running this in the example app so I'll probably link this to my app tomorrow

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by timja** [comment]
I'm finding it quite difficult to test interactively =/

https://github.com/backstage/backstage/blob/master/docs/local-dev/linking-local-packages.md doesn't seem to work for me, does it only work for packages and not plugins?

I've done `yarn tsc && yarn build` and then copied the files in which seems to somewhat work but not always (probably not doing it right).

I was able to test this using the above once and hit and issue with groups that ended in a `)` so I added some additional normalising, haven't tested it interactively, but I've added unit tests that show it fixed.

**Review by Rugvip — COMMENTED** [review]
Looks good to me! 👍 

Any thoughts on this @Fox32?

Will ofc need DCO to be able to merge 😅

**Comment by Fox32** [comment]
In general I see a similar problem on our end with security groups - looks even like MS has changed something here. The `mailNickname` is set so some random characters (not a GUID). But this is also done for groups that are `mailEnabled=true` on our end. Good thing is that every org can set their own `GroupTransformer` to modify it based on their requirements.

I still think that it's fine to merge this.

**Comment by timja** [comment]
> But this is also done for groups that are mailEnabled=true

tbh I didn't check those we barely have any of them I can tweak the PR to always do this?

File: plugins/catalog-backend-module-msgraph/src/microsoftGraph/read.ts [hunk 1]
```diff
@@ -251,6 +251,13 @@ export async function readMicrosoftGraphOrganization(
   return { rootGroup };
 }
 
+function extractGroupName(group: MicrosoftGraph.Group): string {
+  if (group.securityEnabled && !group.mailEnabled) {
```
Comment (timja) [inline plugins/catalog-backend-module-msgraph/src/microsoftGraph/read.ts:255]: @Fox32  ?
```suggestion
  if (group.securityEnabled) {
```

Comment (Fox32) [reply plugins/catalog-backend-module-msgraph/src/microsoftGraph/read.ts:255]: Yeah, let's try it this way.

**Review by Rugvip — APPROVED** [review]
Alright, let's :shipit:, thanks! 👍

**Pull request merged**